script.js

```diff
@@ -1,34 +1,164 @@
 // Exercise 1: Flattening Arrays
 // Write a function to flatten an array by 1 level, i.e. given an array, your function should return an array whose elements are those of the given array and its subarrays, but ignore any subarrays of subarrays. See below for the example, and complete the flatten() function.
-var testArray = [1, [2, 3], [4, [5], 6]];
+// var testArray = [1, [2, 3], [4, [5], 6]];
 
-var flatten = function(arr) {
+// var isArray = function(arr) {
+// 	for (var i = 0; i < arr.length; i++) {
+// 		if (typeof arr[i] === "object") {
+// 			return true;
+// 		}
+// 	}
+// }
 
-}
+// var flattenDeep = function(arr) {
+// 	while (isArray(arr) === true) {
+// 		arr = flatten(arr);
+// 	}
+// 	return arr;
+// }
 
-var result = flatten(testArray);
-console.log(result); // Should print out [1, 2, 3, 4, [5], 6];
+// var flatten = function(arr) {
+// 	var result = [];
+// 	arr.forEach(function(el) {
+// 		if (typeof el !== "number") {
+// 			for (var i = 0; i < el.length; i++) {
+// 				result.push(el[i]);
+// 			}
+// 		} else {
+// 			result.push(el);
+// 		}
+// 	});
+// 	return result;
+// }
+
+// var result = flatten(testArray);
+// var result2 = flattenDeep(testArray);
+// console.log(result); // Should print out [1, 2, 3, 4, [5], 6];
+// console.log(result2);
 
 // Bonus: Flatten the array really hard. No matter how many sub-arrays it has, your function should return an array with no sub-arrays. E.g. flattenDeep(testArray) should return [1, 2, 3, 4, 5, 6].
 // Hint: You will have to find a way to test whether an array has any sub-arrays or not. If it does, your flatten function above should be useful.
 
 
 // Exercise 2: Debugging
 // You'd expect the function below to greet Alan. Why doesn't it? Explain and debug.
-var person1 = {name: "Alan"};
-var person2 = {name: "Amy"};
-var greet = function(person) {
-    if (person === {name: "Alan"}) {
-        console.log("Hello, Alan!");
-    } else {
-        console.log("Hi, Amy!");
-    };
-};
-greet(person1);
+// var person1 = {name: "Alan"};
+// var person2 = {name: "Amy"};
+// var greet = function(person) {
+//     if (person === {name: "Alan"}) {
+//         console.log("Hello, Alan!");
+//     } else {
+//         console.log("Hi, Amy!");
+//     };
+// };
+// greet(person1);
 
-// Explanation: 
-// Write your correct code below.
+// // Explanation: we cannot equate objects to themselves
```

Comment:
This is a little unclear.
var person = {name: "Alan"};
var alan = person;
console.log(person === alan);
This will console.log true...